File: 0235-LowestCommonAncestorOfABinarySearchTree/Solution.java
import java.util.*;

public class Solution {
    
    static TreeNode lca(TreeNode root, TreeNode p, TreeNode q) {
        while (true) {
            if (p.val < root.val && q.val < root.val) {
                root = root.left;
            } else if (p.val > root.val && q.val > root.val) {
                root = root.right;
            } else {
                break;
            }
        }
        return root;
    }

    public static void main(String[] args) {
        System.out.println();
    }

}

File: 0235-LowestCommonAncestorOfABinarySearchTree/solution.py
class Solution:

    def lowest_common_ancestor(self, root, p, q):
        while True:
            if p.val < root.val and q.val < root.val:
                root = root.left
            elif p.val > root.val and q.val > root.val:
                root = root.right
            else:
                break
        return root


if __name__ == "__main__":
    s = Solution()

